checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java

```diff
@@ -138,16 +180,26 @@ private void check(Class clazz) {
 
         final Constructor<?> constructor = constructors[0];
 
-        if (constructor.getParameterTypes().length > 0) {
-            throw new AssertionError(clazz + " has non-default constructor with some parameters");
-        }
-
         constructor.setAccessible(true);
 
         if (!Modifier.isPrivate(constructor.getModifiers())) {
             throw new AssertionError("Constructor of " + clazz + " must be private");
         }
 
+        final Class<?>[] parameterTypes = constructor.getParameterTypes();
+
+        if (parameterTypes.length > 0) {
+            if (parameters == null) {
+                throw new AssertionError(clazz + " has non-default constructor with some parameters");
```

Comment:
This we should not change. That's still the old behavior. The only way parameter is null when we are using `expectedException` or `expectedMessage`